packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceModal.tsx

```diff
@@ -0,0 +1,483 @@
+    title: string;
+    description?: string;
+    selectDescription: string;
+    value: string;
+}
+const StyledSpinner = () => <Spinner size={16} style={{ margin: 12 }} />;
+
+const capitalize = (word: string): string =>
+    word ? `${word.charAt(0).toUpperCase()}${word.slice(1).toLowerCase()}` : '';
+
+const enum SpaceAccessType {
+    PRIVATE = 'private',
+    PUBLIC = 'public',
+}
+const SpaceAccessOptions: AccessOption[] = [
+    {
+        title: 'Restricted access',
+        description: 'Only invited members can access',
+        selectDescription: 'Only invited members can access',
+        value: SpaceAccessType.PRIVATE,
+    },
+    {
+        title: 'Full access',
+        description: 'All project members can access',
+        selectDescription:
+            'All project members can access with their project permissions',
+        value: SpaceAccessType.PUBLIC,
+    },
+];
+
+const enum UserAccessAction {
+    KEEP = 'keep',
+    DELETE = 'delete',
+}
+
+const UserAccessOptions: AccessOption[] = [
+    {
+        title: 'viewer',
+        selectDescription: `This permission has been inherited from user's project access`,
+        value: UserAccessAction.KEEP,
+    },
+    {
+        title: 'No access',
+        selectDescription: `Remove user's access`,
+        value: UserAccessAction.DELETE,
+    },
+];
+
+const renderAccess: ItemRenderer<AccessOption> = (
+    access,
+    { handleClick, handleFocus, modifiers, query },
+) => {
+    if (!modifiers.matchesPredicate) {
+        return null;
+    }
+    return (
+        <MenuItem2
+            multiline={true}
+            active={modifiers.active}
+            disabled={modifiers.disabled}
+            key={access.value}
+            onClick={handleClick}
+            onFocus={handleFocus}
+            text={
+                <>
+                    <AccessSelectTitle>{access.title}</AccessSelectTitle>
+                    <AccessSelectSubtitle>
+                        {access.selectDescription}
+                    </AccessSelectSubtitle>
+                </>
+            }
+        />
+    );
+};
+
+const getUserNameOrEmail = (
+    userUuid: string,
+    organizationUsers: OrganizationMemberProfile[] | undefined,
+) => {
+    const user = organizationUsers?.find(
+        (userAccess) => userAccess.userUuid === userUuid,
+    );
+    if (!user) return userUuid;
+    return user?.firstName ? `${user.firstName} ${user.lastName}` : user.email;
+};
+
+const getInitials = (
+    userUuid: string,
+    organizationUsers: OrganizationMemberProfile[] | undefined,
+) => {
+    const user = organizationUsers?.find(
+        (userAccess) => userAccess.userUuid === userUuid,
+    );
+    if (!user) return userUuid;
+
+    if (user?.firstName) {
+        return user.firstName.substr(0, 1) + user.lastName.substr(0, 1);
+    } else {
+        return user.email.substr(0, 2).toUpperCase();
+    }
+};
+
+const ShareSpaceModal: FC<ShareSpaceProps> = ({ space, projectUuid }) => {
```

Comment:
I feel like this file does more than it supposed to. maybe extract some of the components (in the same folder)